Условие:

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.
2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?
3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?
4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [16]:
from math import factorial
# Создаем функцию для формул сочетания
def combinations (n, k):
    return int(factorial(n) // (factorial(k) * factorial(n - k)))


In [17]:
p = 0.8
k = 85
n = 100
q = 1 - p

#Рассчитаем сочетания
с = combinations (n, k)

#Найдем вероятность того, что стрелок попадет в цель ровно 85 раз
Pn = с * p**k * q**(n-k)
print(Pn)

0.048061793700746355


Ответ: 0.048 или 4.8%

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. 
В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, 
что ни одна из них не перегорит в первый день? 
Какова вероятность, что перегорят ровно две?

In [3]:
p2 = 0.0004
n2 = 5000
m = 0
e = 2.71828
l = p2*n2

#Найдем вероятность того, что ни одна из них не перегорит в первый день
Pm = l**m / factorial(m) * e**-l
print(Pm)

0.13533546530402737


In [4]:
#Найдем вероятность того, что перегорят ровно две?
m2 = 2
Pm2 = l**m2 / factorial(m2) * e**-l
print(Pm2)

0.27067093060805475


Ответ:
Вероятность того, что ни одна из них не перегорит в первый день = 0.135 или 13,5%
Вероятность того, что перегорят ровно две = 0.27 или 27%

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [24]:
n5 = 144
k5 = 70
#Рассчитаем сочетания
с5 = combinations (n5, k5)

p5 = 0.5
q5 = 1 - p5

#Найдем вероятность того, что орел выпадет ровно 70 раз
Pn5 = с5 * p5**k5 * q5**(n5-k5)
print(Pn5)

0.06281178035144776


Ответ:
Вероятность, что орел выпадет ровно 70 раз = 0.063 или 6,3%

4. В первом ящике находится 10 мячей, из которых 7 - белые. 
Во втором ящике - 11 мячей, из которых 9 белых. 
Из каждого ящика вытаскивают случайным образом по два мяча. 
Какова вероятность того, что все мячи белые? 
Какова вероятность того, что ровно два мяча белые?
Какова вероятность того, что хотя бы один мяч белый?

In [41]:
#Найдем вероятность того, что все мечи белые

# Рассчитаем общее количество сочетаний 4 из 21
c1 = combinations (21, 4)
# Рассчитаем количество благоприятных исходов 4 из 16
c2 = combinations (16, 4)

P_o = c2/c1
print(round(P_o, 5))


#Найдем вероятность того, что ровно два мяча белые
n_1 = 10
k_1 = 7

n_2 = 11
k_2 = 9


P_a = k_1/n_1
P_b = k_2/n_2

P_ab = P_a * P_b
print(round(P_ab, 5))


#Найдем вероятность того, что хотя бы 1 мяч - белый

# Рассчитаем общее количество сочетаний 1 из 21
c_1 = combinations (21, 1)
# Рассчитаем количество благоприятных исходов 1 из 16
c_2 = combinations (16, 1)

P_1 = c_2/c_1
print(round(P_1, 5))


0.30409
0.57273
0.7619


Ответ: 
Вероятность того, что все мячи белые = 0.30 или 30%
Вероятность того, что ровно два мяча белые = 0.57 или 57%
Вероятность того, что хотя бы один мяч белый = 0.76 или 76%